src/py21cmfast/wrapper.py

```diff
@@ -2446,6 +2447,9 @@ def run_lightcone(
         If given, must be compatible with init_box. It will merely negate the necessity of
         re-calculating the
         perturb fields. It will also be used to set the redshift if given.
+    coeval_callback : callable, optional
+        User-defined arbitrary function computed on :class:`~Coeval`, at each node redshift.
```

Comment:
Do we really want to do all the node redshifts which can be pretty large? Perhaps, it's better to use an user-defined list with the results being interpolated between the closest two redshifts or (even better) adding the list into `scrollz`. In particular, `scrollz` doesn't always contain the redshifts that are needed for coeval outputs, for instance, we usually need integer redshifts like 6,8,10,15...

Reply:
Yep, that's true I agree, I don't know if it can be easily done, will try

Reply:
following andrei's suggestion, you could also give a new argument like `redshifts_coeval`, and e.g. identify the closest redshift before going into the loop and then only call coeval_callback when you are at those pre-identified points. If you don't like the idea of adding more arguments, maybe the users can define a tolerant z range in the callback function and skip those unnecessary redshifts.